### ISSUE TITLE
Update Nightly Build Process to use Feature Branch or Master

### ISSUE THREAD

**Issue by dstadulis** [lightninglabs/lightning-terminal#818]
Current, the [nightly build process](https://github.com/lightninglabs/lightning-terminal/pull/813/files#top) is set up to build from an [experimental branch](https://github.com/lightninglabs/lightning-terminal/blob/2c6075328f89094ba03437ed6cae197ad7d5365b/.github/workflows/docker.yml#L23), which will eventually be merged into master. To ensure the CD process remains up-to-date, the on-going branch used to build nightlys should be set to either a pre-release branch (say `feature/new-feature`) or based on `master`.

To resolve this issue, the nightly build process should be updated to use one of the following approaches:

1. **Feature Branch**: Configure the nightly build repo to build from a specific feature branch (e.g. `feature/new-feature`) that is regularly updated with the latest changes.
2. **Master Branch**: Update the nightly build repo to build directly from the `master` branch, which will ensure that our builds always reflect the latest codebase.

**TODO:** Once this issue is resolved, the `# TODO(guggero): Change the branch to 'master' once the experimental branch is merged into master.` comment can be removed.

https://github.com/lightninglabs/lightning-terminal/blob/2c6075328f89094ba03437ed6cae197ad7d5365b/.github/workflows/docker.yml#L21-L22
